Add flag for pretty-printing discovery JSON

caxu-rh opened the pull request:
This PR addresses #9.

- Open to alternative naming for the long or short names of the flag. I know of this feature as "pretty-printing" but I suppose it could also be called "formatted" JSON - but I tend to associate `--format` with accepting a template, e.g. `{{ .Foo }}` when I want a program (`kubectl get`, `podman inspect`) to give me some portion of the JSON output (maybe that could be useful too, but a separate feature).
- This defaults to sticking with the one-line, unformatted output. We could also switch this to default to multiline, formatted output, and have a flag to choose unformatted (maybe a `-r`/`--raw-output` flag instead or something similar).
- We add a new boolean flag on `NewManifestJSONProcessorFn`. I considered making this function take a new kind of config/opts struct instead (for future expansion), but I think it could wait for later if/when more options are possible.
- We currently format with indentation of 4 spaces, not sure if this should be 2, or if it should also be configurable.

komish commented:
Oh, and if you end up leaving it - feel free to add a config struct as you see fit.

caxu-rh commented:
- Renamed the option to `--compact` / `-c` for anyone wanting to keep the compacted output. I figure it is still helpful to keep around in case someone is running this in e.g. some CI flow where some external tools might not be available. I think the naming "compact" is more accurate than "raw" (implies that non-raw output is going through some kind of post-processing, which is not the case) - and I realized that `jq`'s `--compact-output` option is closer to what we are doing here (`--raw-output` in `jq` is also omitting escape chars).
- Added a new struct type `NewManifestJSONProcessorFnOptions` - open to naming adjustments. The options in this struct are unique for making a ProcessingFunction that outputs a manifest JSON (not generic to ProcessingFunctions in general). To keep things simple for now, this change includes only the struct (avoiding adding e.g. `WithCompactOutput()` options-style functions).

komish reviewed (approved):
/lgtm
